vumi/persist/riak_manager.py

```diff
@@ -293,16 +293,10 @@ def _search_iteration(self, bucket, query, rows, start):
 
     def real_search(self, modelcls, query, rows=None, start=None):
         rows = 1000 if rows is None else rows
+        start = 0 if start is None else start
```

Comment:
I don't think these other places exist? And if there are a couple of places we can fix them when we find them?

The situation in which the current implementation returns more results is a bit corner case anyway since it requires there to be more results than rows (so more than 1000) by default, but few enough to return in a reasonable time frame.

The cost of raising the exception is having yet another ugly function in the code base that we have to clean up later.